stereotype/fields/compound.py

```diff
@@ -110,13 +111,13 @@ def copy_value(self, value: Any) -> Any:
         item_copy = self.item_field.copy_value
         return [item_copy(item) for item in value]
 
-    def to_primitive(self, value: Any, role: Role = DEFAULT_ROLE) -> Any:
+    def to_primitive(self, value: Any, role: Role = DEFAULT_ROLE, context: ToPrimitiveContextType = None) -> Any:
         if value is None or value is Missing:
             return value
-        if self.item_field.atomic:
+        if not field_method_overriden(self.item_field, 'to_primitive'):
```

Comment:
We would want to do this check once, when the field instance is created, instead of doing it for every `to_primitive` call. So this might need a `Field` instance attribute like `custom_to_primitive` filled out in Field `__init__`.